### PR TITLE
feat: DIDKeyResolver: check key id in did authentication

### LINKED ISSUE #1351: DIDKeyResolver: check key id in did authentication
if you can't find the key id in did public key, try find it in did authentication

https://github.com/hyperledger/aries-framework-go/blob/5531f2c6db0351beec475715ca22162c2b20651c/pkg/doc/verifiable/common.go#L82

### PR DESCRIPTION
**Pull request by fqutishat**
closes #1351

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/doc/verifiable/common.go [hunk 1]
```diff
@@ -85,6 +85,13 @@ func (r *DIDKeyResolver) resolvePublicKey(issuerDID, keyID string) (interface{},
 		}
 	}
 
+	// if key not found in PublicKey try to find it in authentication
```
Comment (troyronda) [inline pkg/doc/verifiable/common.go]: I'm a bit confused, how did you decide priority ordering?

Comment (fqutishat) [reply pkg/doc/verifiable/common.go]: priority not important here because we search by key id

Comment (troyronda) [reply pkg/doc/verifiable/common.go]: I think we will need to revisit.

**Comment by llorllale** [comment]
My reading of the [did-core spec](https://www.w3.org/TR/did-core/#authentication) is that `authentication` is:

> a mechanism by which an entity can prove it is the DID subject.

It goes on to add:

> the verification method indicated by the authentication property of a DID document can only be used to authenticate the DID subject

Is there any reason why these v1 DIDs can't be updated with a publicKey similar to [this](https://w3c-ccg.github.io/did-method-v1/#example-11-did-document-update-request) example (except under `/publicKey`)?

One other point: we should conform to the [did resolution algorithm](https://w3c-ccg.github.io/did-resolution/#dereferencing-algorithm)

**Comment by llorllale** [comment]
It seems like to resolving a key from a DID URL the way this PR wants to is dependent of the MIME type that was attached to the DID document: https://w3c-ccg.github.io/did-resolution/#dereferencing-algorithm-secondary

**Pull request merged**